tm2/pkg/std/tx.go

```diff
@@ -60,6 +60,34 @@ func (tx Tx) ValidateBasic() error {
 	return nil
 }
 
+// IsSponsorTx checks if the transaction is a valid sponsor transaction
+func (tx Tx) IsSponsorTx() bool {
+	// At least two message in the transaction
+	if len(tx.Msgs) <= 1 {
+		return false
+	}
+
+	// The first message type is "no_op"
+	if tx.Msgs[0].Type() != "no_op" {
```

Comment:
this isn't what makes a tx being sponsored or not.

should be:
- if len(msgs) > 1
- for i := 1; i < len(msgs); i++ { if msgs[i].signer != msgs[0].signer }

Reply:
the conditions you mentioned are already in place, I just need to remove the condition where the first message is a `no-op`